vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.cc

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor_util.h"
+
+#include "base/check.h"
+#include "bat/ads/internal/base/logging_util.h"
+#include "bat/ads/internal/base/search_engine/search_engine_results_page_util.h"
+#include "bat/ads/internal/base/search_engine/search_engine_util.h"
+#include "bat/ads/internal/features/text_embedding_features.h"
+#include "bat/ads/internal/locale/locale_manager.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_data.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_processing.h"
+#include "bat/ads/internal/resources/contextual/text_embedding/text_embedding_resource.h"
+#include "bat/ads/internal/resources/language_components.h"
+#include "bat/ads/internal/resources/resource_manager.h"
+#include "bat/ads/internal/tabs/tab_manager.h"
+#include "url/gurl.h"
+
+namespace ads {
+namespace processor {
+
+TextEmbedding::TextEmbedding(resource::TextEmbedding* resource)
+    : resource_(resource) {
+  DCHECK(resource_);
+
+  LocaleManager::GetInstance()->AddObserver(this);
+  ResourceManager::GetInstance()->AddObserver(this);
+  TabManager::GetInstance()->AddObserver(this);
+}
+
+TextEmbedding::~TextEmbedding() {
+  LocaleManager::GetInstance()->RemoveObserver(this);
+  ResourceManager::GetInstance()->RemoveObserver(this);
+  TabManager::GetInstance()->RemoveObserver(this);
+}
+
+bool TextEmbedding::IsEmbeddingEnabled() {
+  return targeting::features::IsTextEmbeddingEnabled();
+}
+
+void TextEmbedding::Process(const std::string& text) {
+  if (!resource_->IsInitialized()) {
+    BLOG(1, "Failed to process token embeddings as resource not initialized");
+    return;
+  }
+
+  const std::string sanitized_text = SanitizeText(text, true);
```

Comment:
might be worth having a comment explaining what the second argument is, like `/*is_html*/`

Reply:
would it be better to have something like:

```
const bool is_html = true;
const std::string sanitized_text = SanitizeText(text, true);
```

?

Reply:
That could also work but maybe it's not necessary. I would just have 
`const std::string sanitized_text = SanitizeText(text, true /*is_html*/)`
it's done also in other parts of brave-core, especially for booleans.
Any preference @tmancey?

Reply:
`/* is_html */ true`, however see my comments about separating concerns and decoupling to `SanitizeText` and `SanitizeHtml`

Reply:
@ptjames when we pass `true` etc. it is preferred to have `/* is_html */ true`

Reply:
Updated